pkg/machinehealthcheck/machinehealthcheck.go

```diff
@@ -41,31 +41,34 @@ var _ = Describe("MachineHealthCheck", framework.LabelMachineHealthChecks, func(
 		var err error
 
 		gatherer, err = framework.NewGatherer()
-		Expect(err).ToNot(HaveOccurred())
+		Expect(err).ToNot(HaveOccurred(), "failed to create a new StateGatherer")
 
 		client, err = framework.LoadClient()
-		Expect(err).ToNot(HaveOccurred())
+		Expect(err).ToNot(HaveOccurred(), "failed to create a new controller-runtime client")
 
 		machineSetParams := framework.BuildMachineSetParams(client, expectedReplicas)
 
 		By("Creating a new MachineSet")
 		machineSet, err = framework.CreateMachineSet(client, machineSetParams)
-		Expect(err).ToNot(HaveOccurred())
+		Expect(err).ToNot(HaveOccurred(), "failed to create a new machineSet resource")
 
 		framework.WaitForMachineSet(client, machineSet.GetName())
 	})
 
 	AfterEach(func() {
 		specReport := CurrentSpecReport()
 		if specReport.Failed() {
-			Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed())
+			err := gatherer.WithSpecReport(specReport).GatherAll()
+			Expect(err).To(Succeed(), "failed to gather spec report")
```

Comment:
Can we merge these back together? And the other cases where these have been split?

Reply:
Yup, sure. Good point.

Reply:
Reverted this change.